pmd-java/src/main/java/net/sourceforge/pmd/lang/java/typeresolution/ClassTypeResolver.java

```diff
@@ -448,8 +449,9 @@ public Object visit(ASTName node, Object data) {
             }
 
             if (i == dotSplitImage.length - 1 && astArguments != null) { // method
-                List<MethodType> methods = getApplicableMethods(previousType, dotSplitImage[i], null,
-                                                                methodArgsArity, accessingClass);
+                List<MethodType> methods = getApplicableMethods(previousType, dotSplitImage[i],
+                                                                new ArrayList<JavaTypeDefinition>(), methodArgsArity,
```

Comment:
we can probably avoid the instantiation here by simply passing `Collections.emptyList()`